lib/qiita/markdown/filters/external_link.rb

```diff
@@ -0,0 +1,45 @@
+require "addressable/uri"
+
+module Qiita
+  module Markdown
+    module Filters
+      class ExternalLink < HTML::Pipeline::Filter
+        def call
+          doc.search("a").each do |anchor|
+            next unless anchor["href"]
+            href = anchor["href"].strip
+            href_host = host_of(href)
+            next unless href_host
+            if external_link?(href_host)
+              anchor["rel"] = "nofollow"
+              anchor["target"] = "_blank"
+            end
+          end
+
+          doc
+        end
+
+        def validate
+          needs :hostname
+        end
+
+        private
+
+        def host_of(url)
+          uri = Addressable::URI.parse(url)
+          uri.host
+        rescue Addressable::URI::InvalidURIError
+          nil
+        end
```

Comment:
Used `addressable` gem to parse the URL including multibyte characters. Also, stopped using regular expression.